app/src/androidTest/java/com/rkm/tasky/repository/implementation/TaskyTaskRepositoryImplTest.kt

```diff
@@ -0,0 +1,156 @@
+package com.rkm.tasky.repository.implementation
+
+import android.content.Context
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.rkm.tasky.database.TaskyDatabase
+import com.rkm.tasky.database.dao.SyncDao
+import com.rkm.tasky.database.dao.TaskDao
+import com.rkm.tasky.database.model.SyncItemType
+import com.rkm.tasky.database.model.SyncUserAction
+import com.rkm.tasky.network.datasource.TaskyTaskRemoteDataSource
+import com.rkm.tasky.repository.mapper.asTask
+import com.rkm.tasky.repository.mapper.asTaskEntity
+import com.rkm.tasky.util.result.Result
+import com.rkm.tasky.utils.json.errorMessageToString
+import com.rkm.tasky.utils.json.taskResponseToPojo
+import com.rkm.tasky.utils.json.taskResponseToString
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runTest
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert.*
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+@RunWith(AndroidJUnit4::class)
+class TaskyTaskRepositoryImplTest {
+
+    private val dispatcher = StandardTestDispatcher()
+    private val id = "97466aaf-262c-4021-8398-28321001123b"
+    private lateinit var server: MockWebServer
+    private lateinit var db: TaskyDatabase
+    private lateinit var syncDataSource: SyncDao
+    private lateinit var retrofit: Retrofit
+    private lateinit var localDataSource: TaskDao
+    private lateinit var remoteDataSource: TaskyTaskRemoteDataSource
+    private lateinit var repository: TaskyTaskRepositoryImpl
+
+    @Before
+    fun setUp() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(context, TaskyDatabase::class.java).build()
+        server = MockWebServer()
+        server.start()
+
+        retrofit = Retrofit.Builder()
+            .baseUrl(server.url("/"))
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        remoteDataSource = retrofit.create(TaskyTaskRemoteDataSource::class.java)
+        syncDataSource = db.syncDao()
+        localDataSource = db.taskDao()
+        repository = TaskyTaskRepositoryImpl(remoteDataSource, localDataSource, syncDataSource, dispatcher)
+    }
+
+    @Test
+    fun getTaskFromRemoteDataSourceSuccess() = runTest(dispatcher) {
+        val response = MockResponse().setResponseCode(200).setBody(taskResponseToString())
+        server.enqueue(response)
+        val result = repository.getTask(id)
+        assertTrue(result is Result.Success)
+        assertEquals((result as Result.Success).data, taskResponseToPojo().asTask())
+    }
+
```

Comment:
Not sure if either of us mentioned this already, but technically, the only thing your tests are testing here is that your TaskyTaskRemoteDataSource correctly wraps response data in Result.Success and correctly puts the error type in Result.Error. The rest doesn't really test anything else from your production code.

And for that you have to write a whole lot of code here. I'd personally recommend to only make use of MockServer when the class that needs it really involves some logic. Imagine a repository that coordinates syncing and caching calls and uses an HttpClient for that. That's way more logic to test than for your simple data source here (which I wouldn't write tests for at all - they'll just cause maintenance effort without ever finding a bug)